Client/Frontend/Browser/UserScriptManager.swift

```diff
@@ -409,6 +409,14 @@ class UserScriptManager {
       alteredSource = alteredSource.replacingOccurrences(of: $0.key, with: $0.value, options: .literal)
     })
     
+    alteredSource = """
+    (function() {
+      if (window.isSecureContext) {
+        \(alteredSource)
+      }
+    })();
+    """
```

Comment:
Lets just add this directly to our ethereum provider JS, since we control that file.  We can keep the other change below for the one we pull from core

Reply:
fixed at [ecb5203](https://github.com/brave/brave-ios/pull/5522/commits/ecb5203100c39a4a42ccfdd5e9585db501017426)